powerstore/resource_snapshot.go

```diff
@@ -258,6 +237,65 @@ func (r *resourceVolumeSnapshot) Read(ctx context.Context, req resource.ReadRequ
 
 // Update - updates volume snapshot resource
 func (r *resourceVolumeSnapshot) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	log.Printf("Started Update")
+
+	//Get plan values
+	var plan models.Snapshot
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	//Get current state
+	var state models.Snapshot
+	diags = req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	if plan.VolumeID.ValueString() != state.VolumeID.ValueString() || plan.VolumeName.ValueString() != state.VolumeName.ValueString() {
+		resp.Diagnostics.AddError(
+			"Error updating volume snapshot resource",
+			"Volume Name or Volume ID cannot be updated")
+		return
+	}
+
+	volModify := r.planToServer(plan)
+
+	//Get volume snapshot ID from state
+	volumeSnapshotID := state.ID.ValueString()
+
+	//Update volume snapshot by calling API
+	_, err := r.client.PStoreClient.ModifyVolume(context.Background(), volModify, volumeSnapshotID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error updating volume snapshot resource",
+			"Could not update volume snapshot "+volumeSnapshotID+": "+err.Error(),
+		)
+		return
+	}
+
+	//Get Volume Snapshot details
+	getRes, err := r.client.PStoreClient.GetSnapshot(context.Background(), volumeSnapshotID)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Error getting snapshot resource after update",
+			"Could not get volume snapshot, unexpected error: "+err.Error(),
+		)
+		return
+	}
+
+	r.updateSnapshotState(&plan, &state, getRes)
+
+	diags = resp.State.Set(ctx, state)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	log.Printf("Successfully done with Update")
```

Comment:
Do we need this ?

Reply:
This is a log message for debug purpose only